testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeDatabricksCloneTableCompatibility.java

```diff
@@ -0,0 +1,274 @@
+                    .containsOnly(expectedRowsV4);
+            assertThat(onTrino().executeQuery("SELECT * FROM delta.default." + baseTable))
+                    .containsOnly(expectedRowsV4);
+            assertThat(onDelta().executeQuery("SELECT * FROM default." + baseTable + " VERSION AS OF 4"))
+                    .containsOnly(expectedRowsV4);
+            assertThat(onDelta().executeQuery("SELECT * FROM default." + clonedTableV4))
+                    .containsOnly(expectedRowsV4);
+            assertThat(onTrino().executeQuery("SELECT * FROM delta.default." + clonedTableV4))
+                    .containsOnly(expectedRowsV4);
+
+            if (partitioned) {
+                List<Row> expectedPartitionRows = ImmutableList.of(row("a"), row("b"));
+                assertThat(onDelta().executeQuery("SELECT b_string FROM default." + baseTable))
+                        .containsOnly(expectedPartitionRows);
+                assertThat(onTrino().executeQuery("SELECT b_string FROM delta.default." + baseTable))
+                        .containsOnly(expectedPartitionRows);
+                assertThat(onDelta().executeQuery("SELECT b_string FROM default." + baseTable + " VERSION AS OF 3"))
+                        .containsOnly(expectedPartitionRows);
+                assertThat(onDelta().executeQuery("SELECT b_string FROM default." + clonedTableV3))
+                        .containsOnly(expectedPartitionRows);
+                assertThat(onTrino().executeQuery("SELECT b_string FROM delta.default." + clonedTableV3))
+                        .containsOnly(expectedPartitionRows);
+            }
+
+            onDelta().executeQuery("INSERT INTO default." + clonedTableV4 + " VALUES (3, \"c\", 3)");
+            onTrino().executeQuery("INSERT INTO delta.default." + clonedTableV4 + " VALUES (4, 'd', 4)");
+
+            List<Row> expectedRowsV5 = ImmutableList.of(row(1, "a", null), row(2, "b", 3), row(3, "c", 3), row(4, "d", 4));
+            assertThat(onTrino().executeQuery("SELECT * FROM delta.default." + clonedTableV4))
+                    .containsOnly(expectedRowsV5);
+            assertThat(onDelta().executeQuery("SELECT * FROM default." + clonedTableV4))
+                    .containsOnly(expectedRowsV5);
+        }
+        finally {
+            dropDeltaTableWithRetry("default." + baseTable);
+            dropDeltaTableWithRetry("default." + clonedTableV1);
+            dropDeltaTableWithRetry("default." + clonedTableV2);
+            dropDeltaTableWithRetry("default." + clonedTableV3);
+            dropDeltaTableWithRetry("default." + clonedTableV4);
+        }
+    }
+
+    @DataProvider
+    public Object[][] shallowTrueFalse()
+    {
+        return new Object[][] {{"SHALLOW", true}, {"SHALLOW", false}};
+    }
+
+    @DataProvider
+    public Object[][] deepTrueFalse()
+    {
+        return new Object[][] {{"DEEP", true}, {"DEEP", false}};
+    }
```

Comment:
No need to define data providers here. We can inline `SHALLOW` & `DEEP` and use `DataProviders#trueFalse`.